applications/matter_bridge/src/bridged_device_types/ble_environmental_data_provider.cpp

```diff
@@ -0,0 +1,268 @@
+
+bt_gatt_read_params BleEnvironmentalDataProvider::sHumidityReadParams{};
+
+BleEnvironmentalDataProvider *GetProvider(bt_conn *conn)
+{
+	BLEBridgedDevice *device = BLEConnectivityManager::Instance().FindBLEBridgedDevice(conn);
+	VerifyOrReturnValue(device, nullptr);
+	return reinterpret_cast<BleEnvironmentalDataProvider *>(device->mProvider);
+}
+
+bt_uuid *BleEnvironmentalDataProvider::GetServiceUuid()
+{
+	return sServiceUuid;
+}
+
+uint8_t BleEnvironmentalDataProvider::GattTemperatureNotifyCallback(bt_conn *conn, bt_gatt_subscribe_params *params,
+								    const void *data, uint16_t length)
+{
+	BleEnvironmentalDataProvider *provider = GetProvider(conn);
+	VerifyOrExit(provider, );
+	VerifyOrExit(data, );
+	VerifyOrExit(length == sizeof(mTemperatureValue), );
+
+	/* Save data received in notification. */
+	memcpy(&provider->mTemperatureValue, data, length);
+	DeviceLayer::PlatformMgr().ScheduleWork(NotifyTemperatureAttributeChange, reinterpret_cast<intptr_t>(provider));
+
+exit:
+
+	return BT_GATT_ITER_CONTINUE;
+}
+
+uint8_t BleEnvironmentalDataProvider::GattHumidityNotifyCallback(bt_conn *conn, bt_gatt_subscribe_params *params,
+								 const void *data, uint16_t length)
+{
+	BleEnvironmentalDataProvider *provider = GetProvider(conn);
+	VerifyOrExit(provider, );
+	VerifyOrExit(data, );
+	VerifyOrExit(length == sizeof(mHumidityValue), );
+
+	/* Save data received in notification. */
+	memcpy(&provider->mHumidityValue, data, length);
+	DeviceLayer::PlatformMgr().ScheduleWork(NotifyHumidityAttributeChange, reinterpret_cast<intptr_t>(provider));
+
+exit:
+
+	return BT_GATT_ITER_CONTINUE;
+}
+
+void BleEnvironmentalDataProvider::Init()
+{
+	/* Configure subscription for the temperature characteristic */
+	mGattTemperatureSubscribeParams.ccc_handle = mCccTemperatureHandle;
+	mGattTemperatureSubscribeParams.value_handle = mTemperatureCharacteristicHandle;
+	mGattTemperatureSubscribeParams.value = BT_GATT_CCC_NOTIFY;
+	mGattTemperatureSubscribeParams.notify = BleEnvironmentalDataProvider::GattTemperatureNotifyCallback;
+
+	/* Configure subscription for the humidity characteristic */
+	mGattHumiditySubscribeParams.ccc_handle = mCccHumidityHandle;
+	mGattHumiditySubscribeParams.value_handle = mHumidityCharacteristicHandle;
+	mGattHumiditySubscribeParams.value = BT_GATT_CCC_NOTIFY;
+	mGattHumiditySubscribeParams.notify = BleEnvironmentalDataProvider::GattHumidityNotifyCallback;
+
+	Subscribe();
+}
+
+void BleEnvironmentalDataProvider::NotifyUpdateState(chip::ClusterId clusterId, chip::AttributeId attributeId,
+						     void *data, size_t dataSize)
+{
+	if (mUpdateAttributeCallback) {
+		mUpdateAttributeCallback(*this, clusterId, attributeId, data, dataSize);
+	}
+}
+
+CHIP_ERROR BleEnvironmentalDataProvider::UpdateState(chip::ClusterId clusterId, chip::AttributeId attributeId,
+						     uint8_t *)
+{
+	LOG_INF("Updating state of the BleEnvironmentalDataProvider, cluster ID: %u, attribute ID: %u. Dropping, currently not supported.",
```

Comment:
I think this log is not needed, it's a common situation that write may not supported for some device type.

Reply:
It's good for debugging but I'm fine with removing it. But to be consistent we should do the same in the simulated temperature and humidity data providers.

Reply:
I didn't even know we have such logs there too and I thought this provider will do it in a different way. Then I leave the decision to you, I'm ok with both leaving it or removing everywhere.